File: src/offer10-2.Jump/jump.cpp
class Solution {
public:
    int jumpFloor(int number) {
        int p = 1;
        int q = 2;
        if (number <= 2){
            return number;
        }
        for(int i = 3; i <= number; i++){
            q = p + q;
            p = q - p;
        }
        return q;
    }
};

File: src/offer10-2.Jump/jump.java
public class Solution {
    public int JumpFloor(int target) {
        int step1 = 1;
        int step2 = 2;
        if(target <= 2){
            return target;
        }
        for(int i = 3; i <= target; i++){
            step2 += step1;
            step1 = step2 - step1;
        }
        return step2;
    }
}